microbench/src/main/java/com/joyent/http/signature/BenchmarkSigner.java

```diff
@@ -24,30 +24,29 @@
  * information on JMH and running benchmarks.  In general one would
  * "package" this module, and run `java -jar target/benchmark.jar`. */
 @State(Scope.Thread)
-public class BenchmarkSigner {
-    private String testKeyFingerprint;
-    private KeyPair keyPair;
-    private Signer signer;
-    private String verifyNow;
-    private String verifyHeader;
+@SuppressWarnings({"checkstyle:javadocmethod", "checkstyle:javadoctype", "checkstyle:javadocvariable",
+            "checkstyle:visibilitymodifier"})
+public abstract class BenchmarkSigner {
+    protected KeyPair keyPair;
+    protected String testKeyFingerprint;
+    protected Signer signer;
+    protected String verifyNow;
+    protected String verifyHeader;
 
-    @Param({"vanilla", "native"})
-    private String signType;
+    @Param({"SHA1", "SHA256", "SHA512"})
+    private String hash;
+
+
+    @Param({"stdlib", "native.jnagmp"})
+    private String providerCode;
+
+    public abstract String getKeyCode();
 
     @Setup
     public void setup() throws IOException {
-        switch (signType) {
-        case "vanilla":
-            signer = new Signer(false);
-            break;
-        case "native":
-            signer = new Signer(true);
-            break;
-        default:
-            throw new IllegalArgumentException();
-        }
-        testKeyFingerprint = SignerTestUtil.testKeyFingerprint;
-        keyPair = SignerTestUtil.testKeyPair(signer);
+        testKeyFingerprint = SignerTestUtil.testKeyFingerprint(getKeyCode());
+        keyPair = SignerTestUtil.testKeyPair(getKeyCode());
+        signer = new Signer.Builder(keyPair).hash(hash).providerCode(providerCode).build();
```

Comment:
JMH doesn't seem to have a "nice" way to do output, so it ends up looking something like:

```
# Run progress: 0.00% complete, ETA 04:53:20
# Fork: 1 of 10
# Warmup Iteration   1: 
#Signature-->Provider: SUN
1103.294 ops/s
```

And I needed to re-add `getSignature` to `Signer` as package-private.  I don't think that's a big deal.